### PR TITLE
Make reproduction CI test

### PR DESCRIPTION
**Pull request by GuardianDll**
#### Summary
None
#### Purpose of change
There is no reason to hold back #76266 because i can't get what's wrong with the test i made
#### Describe the solution
Move the test from #76266 here
#### Testing
image
#### Additional context
Relies on #76266 for obv reasons, and relies on #76264 because otherwise i'd need to do shenanigans with unloading and loading monster back

### CODE REVIEW

**Comment by mqrause** [comment]
`try_reproduce` doesn't actually spawn monsters, it just tells the map to spawn them on the next occasion. So calling `spawn_monsters` on the map after `try_reproduce` should probably fix it.

**Comment by GuardianDll** [comment]
what about failing `monsters_spawn_egg_itemgroups`? this one spawns items, and... oh right, it's probably because `spawn_items()` works only outside of bub. How would i do it then? run `try_reproduce()`, then teleport avatar afar, and then teleport them back?

**Comment by mqrause** [comment]
I was looking at [this run](https://github.com/CleverRaven/Cataclysm-DDA/actions/runs/10754561461/job/29825690506), which didn't fail for eggs.

**Comment by GuardianDll** [comment]
🤔 
i'll come back to this PR when previous two will be merged

File: data/mods/TEST_DATA/monsters.json [hunk 1]
```diff
@@ -427,5 +427,50 @@
     "dodge": 2,
     "bleed_rate": 60,
     "armor": { "bash": 8, "cut": 10, "electric": 1 }
+  },
+  {
+    "id": "mon_dummy_reproducer_eggs",
+    "type": "MONSTER",
+    "name": { "str": "debug rerpoducer (eggs)", "str_pl": "debug rerpoducers (eggs)" },
```
Comment (Night-Pryanik) [inline data/mods/TEST_DATA/monsters.json:434]:
```suggestion
    "name": { "str": "debug reproducer (eggs)", "str_pl": "debug reproducers (eggs)" },
```

File: data/mods/TEST_DATA/monsters.json [hunk 2]
```diff
@@ -427,5 +427,50 @@
...
+    "color": "white",
+    "symbol": "1",
+    "reproduction": { "baby_type": { "baby_egg": "egg_chicken" }, "baby_count": 1, "baby_timer": 1 },
+    "baby_flags": [ "SPRING", "SUMMER", "AUTUMN", "WINTER" ]
+  },
+  {
+    "id": "mon_dummy_reproducer_egg_group",
+    "copy-from": "mon_dummy_reproducer_eggs",
+    "type": "MONSTER",
+    "name": { "str": "debug rerpoducer (egg group)", "str_pl": "debug rerpoducers (egg group)" },
```
Comment (Night-Pryanik) [inline data/mods/TEST_DATA/monsters.json:450]:
```suggestion
    "name": { "str": "debug reproducer (egg group)", "str_pl": "debug reproducers (egg group)" },
```

File: data/mods/TEST_DATA/monsters.json [hunk 3]
```diff
@@ -427,5 +427,50 @@
...
+    "color": "green",
+    "symbol": "2",
+    "reproduction": { "baby_type": { "baby_egg_group": "book_gunmags" }, "baby_count": 1, "baby_timer": 1 },
+    "baby_flags": [ "SPRING", "SUMMER", "AUTUMN", "WINTER" ]
+  },
+  {
+    "id": "mon_dummy_reproducer_mon",
+    "copy-from": "mon_dummy_reproducer_eggs",
+    "type": "MONSTER",
+    "name": { "str": "debug rerpoducer (monster)", "str_pl": "debug rerpoducers (monster)" },
```
Comment (Night-Pryanik) [inline data/mods/TEST_DATA/monsters.json:460]:
```suggestion
    "name": { "str": "debug reproducer (monster)", "str_pl": "debug reproducers (monster)" },
```

File: data/mods/TEST_DATA/monsters.json [hunk 4]
```diff
@@ -427,5 +427,50 @@
...
+    "color": "cyan",
+    "symbol": "3",
+    "reproduction": { "baby_type": { "baby_monster": "mon_dog_bull" }, "baby_count": 1, "baby_timer": 1 },
+    "baby_flags": [ "SPRING", "SUMMER", "AUTUMN", "WINTER" ]
+  },
+  {
+    "id": "mon_dummy_reproducer_mon_group",
+    "copy-from": "mon_dummy_reproducer_eggs",
+    "type": "MONSTER",
+    "name": { "str": "debug rerpoducer (mongroup)", "str_pl": "debug rerpoducers (mongroup)" },
```
Comment (Night-Pryanik) [inline data/mods/TEST_DATA/monsters.json:470]:
```suggestion
    "name": { "str": "debug reproducer (mongroup)", "str_pl": "debug reproducers (mongroup)" },
```

**Pull request merged**